specification/communication/data-plane/Chat/preview/2024-03-15-preview/communicationserviceschat.json

```diff
@@ -0,0 +1,2068 @@
+          "type": "string",
+          "example": "1591137790240"
+        },
+        "readOn": {
+          "format": "date-time",
+          "description": "The time at which the message was read. The timestamp is in RFC3339 format: `yyyy-MM-ddTHH:mm:ssZ`.",
+          "type": "string",
+          "example": "2020-10-30T10:50:50Z"
+        }
+      }
+    },
+    "ChatMessageReadReceiptsCollection": {
+      "description": "A paged collection of chat message read receipts.",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of chat message read receipts.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ChatMessageReadReceipt"
+          }
+        },
+        "nextLink": {
+          "description": "If there are more chat message read receipts that can be retrieved, the next link will be populated.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "CommunicationError": {
+      "$ref": "../../../Common/stable/2023-11-15/common.json#/definitions/CommunicationError"
+    },
+    "CommunicationErrorResponse": {
+      "$ref": "../../../Common/stable/2023-11-15/common.json#/definitions/CommunicationErrorResponse"
+    },
+    "SendReadReceiptRequest": {
+      "description": "Request payload for sending a read receipt.",
+      "required": [
+        "chatMessageId"
+      ],
+      "type": "object",
+      "properties": {
+        "chatMessageId": {
+          "description": "Id of the latest chat message read by the user.",
+          "type": "string",
+          "example": "1592435762364"
+        }
+      }
+    },
+    "ChatMessageType": {
+      "description": "The chat message type.",
+      "enum": [
+        "text",
+        "html",
+        "topicUpdated",
+        "participantAdded",
+        "participantRemoved"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ChatMessageType",
+        "modelAsString": true
+      },
+      "example": "html"
+    },
+    "SendChatMessageRequest": {
+      "description": "Details of the message to send.",
+      "required": [
+        "content"
+      ],
+      "type": "object",
+      "properties": {
+        "content": {
+          "description": "Chat message content.",
+          "type": "string",
+          "example": "<p>Come one guys, lets go for lunch together.</p>"
+        },
+        "senderDisplayName": {
+          "description": "The display name of the chat message sender. This property is used to populate sender name for push notifications.",
+          "type": "string",
+          "example": "Bob Admin"
+        },
+        "type": {
+          "$ref": "#/definitions/ChatMessageType"
+        },
+        "metadata": {
+          "description": "Message metadata.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
```

Comment:
"type: string" here means that properties of "metadata" can only have string values -- no numbers or booleans or objects. Is that what you want?

Reply:
Would need to have number and boolean as string if want to use in medatadata